src/lib/server/models.ts

```diff
@@ -1,10 +1,22 @@
-import { HF_ACCESS_TOKEN, MODELS, OLD_MODELS } from "$env/static/private";
+import { HF_ACCESS_TOKEN, OPENAI_API_KEY, MODELS, OLD_MODELS } from "$env/static/private";
 import type { ChatTemplateInput, WebSearchQueryTemplateInput } from "$lib/types/Template";
 import { compileTemplate } from "$lib/utils/template";
 import { z } from "zod";
 
 type Optional<T, K extends keyof T> = Pick<Partial<T>, K> & Omit<T, K>;
 
+const openAICompatibleEndpoint = z.object({
+	host: z.literal("openai-compatible"),
+	baseURL: z.string().url().default("https://api.openai.com/v1"),
+	apiKey: z
+		.string()
+		.min(1)
+		.default(OPENAI_API_KEY ?? "sk-"),
```

Comment:
Actually `OpenAI` instance would throw errors (by design) if there is no key provided, so we still need to provide `sk-` in some way as default to prevent this behaviour. https://github.com/huggingface/chat-ui/pull/452/commits/d9509668bbe4550ee331d31f332c814b748c68f1